framework/src/bcs/HFEMDirichletBC.C

```diff
@@ -0,0 +1,115 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "HFEMDirichletBC.h"
+
+registerMooseObject("MooseApp", HFEMDirichletBC);
+
+InputParameters
+HFEMDirichletBC::validParams()
+{
+  InputParameters params = ArrayLowerDIntegratedBC::validParams();
+  params.addParam<RealEigenVector>("value", "Value of the BC");
+  params.addCoupledVar("uhat", "The coupled variable");
+  params.addParam<bool>(
+      "for_pjfnk", false, "True to add non-zero entries on lambda diagonal block");
+  params.addClassDescription("Imposes the Dirichlet BC with HFEM.");
+  return params;
+}
+
+HFEMDirichletBC::HFEMDirichletBC(const InputParameters & parameters)
+  : ArrayLowerDIntegratedBC(parameters),
+    _value(isParamValid("value") ? getParam<RealEigenVector>("value")
+                                 : RealEigenVector::Zero(_count)),
+    _for_pjfnk(getParam<bool>("for_pjfnk")),
```

Comment:
I _really_ don't like this. Is there not an alternative? At least a better name.

Reply:
I should have removed this parameter. I was planning to add nonzero diagonal jacobian for PJFNK with default preconditioning matrix, but that does not work very well because lots of times the preconditioning matrix would still be singular.